xmtp/src/account.rs

```diff
@@ -34,8 +83,87 @@ impl<'de> Deserialize<'de> for VmacAccount {
     where
         D: serde::de::Deserializer<'de>,
     {
-        let pickle: AccountPickle = Deserialize::deserialize(deserializer)?;
-        let account = Account::from_pickle(pickle);
+        let pickle: OlmAccountPickle = Deserialize::deserialize(deserializer)?;
+        let account = OlmAccount::from_pickle(pickle);
+
         Ok(Self::new(account))
     }
 }
+
+#[derive(Serialize, Deserialize)]
+pub struct Account {
+    keys: VmacAccount,
+    assoc: Association,
+}
```

Comment:
VmacAccount remains as an wrapper around the OlmAccount providing abstraction + Serialization

Account integrates all the signing and application specific context with the keys. In most cases Account replaces VmacAccount

Reply:
I like this